[Vanessa Kang] iP

vanessa-kang opened the pull request.

vanessa-kang committed:
Added Level-0 increment (wk2)

vanessa-kang committed:
Added Level-1 increment (wk3)

vanessa-kang committed:
Added Level-2 increment (wk3): Add, List

vanessa-kang committed:
Added Level-3 and A-Classes extension (wk3): Mark as Done, implement Task class

vanessa-kang committed:
Added Level-4 and A-Inheritance extension (wk4): ToDos, Events, Deadlines

EyoWeiChin commented on src/main/java/Duke.java:
```diff
@@ -1,59 +1,122 @@
+                newObj = new Todo((convertToString(descArr)));
+                break;
+            }
+            case("deadline"): {
```
Should the case statements be on the same indentation with the switch statement?

EyoWeiChin commented on src/main/java/Duke.java:
```diff
@@ -1,59 +1,122 @@
 
 public class Duke {
 
     public static String indent = "     ";
```
I think this could be changed to a constant variable by using 'final'?
example: 
public static final INDENT_SPACE = "     ";

Zhi-You commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,24 @@
+    // added attribute
+    private String by;
+
+    public Deadline(String desc, String by) {
```
Perhaps a more intuitive variable name here for 'desc'?

Zhi-You commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,126 @@
+    // helper function to convert string array to string
+    // not just string rep like what Arrays.toString(strArr) does
+    public static String convertToString(String[] strArr) {
+//        StringBuilder builder = new StringBuilder();
+//        for(String s : strArr) {
+//            builder.append(s + " ");
+//        }
+//        return builder.toString();
```
Perhaps these can be extracted out if not in use? You can consider storing these comments somewhere accessible if you want to refer to it in future, but I think it would be better to extract them out now to make the commits in your pull request more readable!

Zhi-You commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,126 @@
+        else {
+            allTasks.get(num - 1).setIsDone(true);
+            System.out.println(indent + "Nice! I've marked this task as done:");
+            System.out.println(indent + "[\u2713] " + allTasks.get(num - 1).getTask());
```
Should the tick be made into a constant to avoid having magic literals in your code? I noticed that there are a few more occurrences of this in your code elsewhere.

Zhi-You commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,126 @@
+
+        switch(inputArr[0]) {
+            case("todo"): {
+                String[] descArr = Arrays.copyOfRange(inputArr, 1, inputArr.length);
```
Perhaps you can use a more intuitive name for "descArr"? I noticed the same issue in several other variables too.

Zhi-You commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,126 @@
+    }
+
+
     public static void main(String[] args) {
```
Perhaps you can consider refactoring some parts over here? Considering this is the main method, I think it would be better if it could be made more readable such that readers can quickly understand what behaviour the code will give when the program is ran.

Zhi-You commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,126 @@
+
+            userCmd = inputArr[0].toLowerCase();
+
+            switch (userCmd) {
```
Should the case statements be indented to the same extent as the switch () statement?

Zhi-You commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,25 @@
+        this.at = at;
+    }
+
+    // override method
```
Should this be @override instead? If you think so too, then it would be applicable to the other overridden methods that you have!

Zhi-You commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,126 @@
+
+        Task newObj;
+
+        switch(inputArr[0]) {
```
Perhaps you might want to create a variable for inputArr[0], such that it would be clearer for the readers and yourself as to what you are checking in your switch statement?

Zhi-You reviewed (commented):
Generally I feel that the code is all right and you have followed coding practices closely too.
However, I feel that more refactoring can be done (SLAP), names could be named more intuitively, magic numbers can be avoided by declaring constants and comments can be added to explain to readers and to yourself what that section of the code does!

EyoWeiChin reviewed (commented):
I think the code looks good aside from some minor coding standard issues!

EyoWeiChin commented on src/main/java/Duke.java:
```diff
@@ -1,59 +1,122 @@
+        if (num > allTasks.size() || num == 0) {
             System.out.println(indent + "Sorry, this task does not exist!");
         }
         else {
```
Should this 'else' be on the same line as the closing brace of the 'if' statement?

EyoWeiChin commented on src/main/java/Duke.java:
```diff
@@ -1,59 +1,122 @@
+           - deadline/event does not have "/by" or "/at"
+        */
+
+        Task newObj;
```
The Task name newObj is not very obvious later on, perhaps it could benefit code readability if it was changed to newTask?

EyoWeiChin commented on src/main/java/Duke.java:
```diff
@@ -1,59 +1,122 @@
+
     public static void main(String[] args) {
 
         System.out.println(indent + "Hello! I'm Duke\n" + indent + "What can I do for you?\n");
```
To add on to what Zhi-You have pointed out, perhaps this line could be refactored out as a method? This would help the code adhere closer to SLAP

EyoWeiChin commented on src/main/java/Duke.java:
```diff
@@ -1,59 +1,122 @@
-    public static void updateTaskAsDone(int num) {
-        if (num >= allTasks.size()) {
+    // print newly added task
+    public static void printNewlyAddedTask(Task task) {
+        System.out.println(indent + "Got it. I've added this task: "
+                + System.lineSeparator()
+                + indent + indent + task
+                + System.lineSeparator()
+                + indent + "Now you have " + allTasks.size() + " tasks in the list.");
+    }
```
I like how this was method was formatted, very nice!

vanessa-kang committed:
Added Level 5 (wk5): Handle Errors. Just the basic two for now

vanessa-kang committed:
Added exception handling for 'done' command

vanessa-kang committed:
Revert "Added exception handling for 'done' command"

This reverts commit 381f72bf629944db1fda48ba8f16fa220b8ac292.

vanessa-kang committed:
test

vanessa-kang committed:
Merging Level-5 and A-Exceptions into master

vanessa-kang committed:
Added Level-6 and A-Collections (wk6): delete, ArrayList

vanessa-kang committed:
Added Level-7: save. Haven't handled the cases where file or dir doesn't exist at the start yet.

vanessa-kang committed:
Handled cases where file or dir does not initially exist.

vanessa-kang committed:
removed data dir, updated gitignore

vanessa-kang committed:
Merge branch 'branch-Level-6'
Adds delete functionality

vanessa-kang committed:
Merge branch 'branch-Level-7'

vanessa-kang committed:
Refactored code to adhere more closely to Coding Standard and Code Quality. Also packaged as JAR file

Samuel787 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,29 @@
+public class Event extends Task {
+
+    // added attribute
```
Comments are meant to explain the code

Samuel787 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,292 @@
+
+            switch (userKeyWord) {
+            case("todo"):
+            case("deadline"):
```
Empty case bodies may be unnecessary

Samuel787 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,292 @@
+            userKeyWord = inputArr[0].toLowerCase();
+
+            switch (userKeyWord) {
+            case("todo"):
```
Having no break statements in cases can lead to execution flowing into default block and introduce bugs.

Samuel787 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,292 @@
+        }
+    }
+
+    // print array of Task objects
```
It is recommended that comments describing functions follow present tense.

Samuel787 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,292 @@
+    public static final String dirPath = "./data";
+    public static final String dataFilePath = "./data/duke.txt";
+
+    // init array of Task objects
```
Comment here might be unnecessary

Samuel787 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,292 @@
+        Scanner s = new Scanner(fr);
+        while (s.hasNext()) {
+            String[] taskStrArr = s.nextLine().split(">");
+//            for (int i = 0; i < taskStrArr.length; i++) {
+//                taskStrArr[i] = taskStrArr[i].trim();
+//            }
+
+            // strArr[0] - type of Task (T/D/E)
+            // strArr[1] - whether task isDone (0/1)
+            // strArr[2] - task description
+            // strArr[3] - by/at (for deadline/event only)
```
You can consider removing old code from the codebase instead of commenting it when making a PR

Samuel787 reviewed (commented):
Overall, good coding standard and code quality. Well done!

Samuel787 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,292 @@
+    }
+
+    // delete task from allTasks
+    public static void deleteTask(int num) {
```
You have added all the different commands as methods in your Duke.java. Please consider refactoring your code to include more OOP.

Samuel787 reviewed (commented):
You have added all the different commands as methods in your Duke.java. Please consider refactoring your code to include more OOP.

vanessa-kang committed:
Added A-MoreOOP increment.

vanessa-kang committed:
Added Level-9 increment: Find

vanessa-kang committed:
Further refactored TaskList and Ui classes.

vanessa-kang committed:
Added A-JavaDoc increment + some minor cleanup.

vanessa-kang committed:
Merge pull request #1 from vanessa-kang/branch-Level-9

Merge Level-9 (find) increment from Branch-Level-9 to master

vanessa-kang committed:
Merge pull request #2 from vanessa-kang/branch-A-JavaDoc

Merge A-JavaDoc increment (+ some minor cleanup) from branch-A-JavaDoc to master

vanessa-kang committed:
Added A-UserGuide increment.

vanessa-kang committed:
Set theme jekyll-theme-tactile

vanessa-kang committed:
Added help command to list available commands.
Also did minor cleanup of Ui class.

vanessa-kang committed:
Merge branch 'master' of https://github.com/vanessa-kang/ip

vanessa-kang committed:
Further refactored Ui class.

vanessa-kang committed:
Updated README to include help command.

vanessa-kang committed:
Modified list of help commands printed.

vanessa-kang committed:
Fixed bug which was causing file loading error at program restart.
Deadlines/events without a date were being allowed to be added into the task list.
This did not raise any issues while saving and exiting the program,
but threw an IndexOutOfBoundsException at program restart and file data loading.
Fixed by disallowing date-less deadlines/events from being added into the task list in the first place.

vanessa-kang committed:
Improved file IO UI messages, added more JavaDoc comments.
On program startup, file IO messages for file creation and file loading have been separated, in order to make the current file status clearer to the user.

vanessa-kang committed:
Fixed bug where uppercase inputs were throwing unhandled exceptions.
Fixed by converting all inputs to lowercase before parsing.

vanessa-kang committed:
Minor code quality improvements.
